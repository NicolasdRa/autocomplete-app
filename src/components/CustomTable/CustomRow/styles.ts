import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		root: {
			'& > *': {
				borderBottom: 'unset',
			},
			'&:hover': {
				cursor: 'pointer',
			},
		},

		productImg: {
			marginLeft: '2rem',
			height: '4rem',
		},

		infoContainer: {
			display: 'flex',
			flexDirection: 'column',
			alignContent: 'left',
		},

		primaryInfo: {
			fontSize: '.8rem',
			padding: '1rem 1rem .5rem 0',
		},

		secondaryInfo: {
			fontSize: '.7rem',
			padding: '0 1rem 1rem 0',
		},
	}),
);
