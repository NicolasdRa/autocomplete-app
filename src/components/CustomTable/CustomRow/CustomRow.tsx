import { useState } from 'react';

import {
	Box,
	Collapse,
	IconButton,
	TableRow,
	TableCell,
	Typography,
} from '@material-ui/core';
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';

import { RowExpandedContent } from '../RowExpandedContent/RowExpandedContent';

import { useStyles } from './styles';

interface Props {
	data: {
		title: string;
		gtin: string;
		gender: string;
		price: string;
		sale_price: string;
		image_link: string;
		additional_image_link: string;
	};
}

export const CustomRow: React.FC<Props> = ({ data }) => {
	const classes = useStyles();
	const [open, setOpen] = useState(false);

	const {
		title,
		gtin,
		gender,
		price,
		sale_price,
		image_link,
		additional_image_link,
	} = data;

	return (
		<>
			<TableRow
				className={classes.root}
				hover
				onClick={() => setOpen(!open)}>
				<TableCell className={classes.firstCell}>
					<img
						loading='lazy'
						src={image_link}
						alt={title}
						className={classes.productImg}
					/>
				</TableCell>
				<TableCell className={classes.secondCell}>
					<Typography className={classes.primaryInfo}>
						{title}
					</Typography>
				</TableCell>
				<TableCell>
					<Typography className={classes.primaryInfo}>
						{gtin}
					</Typography>
				</TableCell>
				<TableCell>
					<Typography className={classes.primaryInfo}>
						{gender}
					</Typography>
				</TableCell>
				<TableCell>
					<Typography className={classes.primaryInfo}>
						{price}
					</Typography>
				</TableCell>
				<TableCell>
					<Typography className={classes.primaryInfo}>
						{sale_price}
					</Typography>
				</TableCell>
				<TableCell>
					<IconButton
						aria-label='expand row'
						size='small'
						onClick={() => setOpen(!open)}>
						{open ? (
							<KeyboardArrowUpIcon />
						) : (
							<KeyboardArrowDownIcon />
						)}
					</IconButton>
				</TableCell>
			</TableRow>
			<TableRow>
				<TableCell
					style={{
						padding: 0,
					}}
					colSpan={12}>
					<Collapse in={open} timeout='auto' unmountOnExit>
						<Box>
							<RowExpandedContent data={additional_image_link} />
						</Box>
					</Collapse>
				</TableCell>
			</TableRow>
		</>
	);
};
