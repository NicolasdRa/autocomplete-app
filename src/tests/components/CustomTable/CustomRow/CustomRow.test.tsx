import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { CustomRow } from '../../../../components/CustomTable/CustomRow/CustomRow';

const mockData = {
	title: 'test object',
	gtin: '1',
	gender: 'unisex',
	price: '1 euro',
	sale_price: '0.5 euro',
	image_link: 'testImg.jpg',
	additional_image_link: 'testImg1.jpg, testImg2.jpg',
};

describe('tests on <CustomRow/>', () => {
	const wrapper = mount(<CustomRow data={mockData} />);
	const button = wrapper.find('button');

	test('should render <CustomRow/> correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('expanded content should mount on click on btn', () => {
		button.simulate('click');
		expect(wrapper.find('RowExpandedContent').exists()).toBe(true);
	});
});
