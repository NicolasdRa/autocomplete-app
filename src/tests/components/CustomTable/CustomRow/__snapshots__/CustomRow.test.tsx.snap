// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <CustomRow/> should render <CustomRow/> correctly 1`] = `
Array [
  <tr
    className="MuiTableRow-root makeStyles-root-1 MuiTableRow-hover"
    onClick={[Function]}
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root makeStyles-firstCell-2"
    >
      <img
        alt="test object"
        className="makeStyles-productImg-3"
        loading="lazy"
        src="testImg.jpg"
      />
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root makeStyles-secondCell-4"
    >
      <p
        className="MuiTypography-root makeStyles-primaryInfo-5 MuiTypography-body1"
      >
        test object
      </p>
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root"
    >
      <p
        className="MuiTypography-root makeStyles-primaryInfo-5 MuiTypography-body1"
      >
        1
      </p>
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root"
    >
      <p
        className="MuiTypography-root makeStyles-primaryInfo-5 MuiTypography-body1"
      >
        unisex
      </p>
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root"
    >
      <p
        className="MuiTypography-root makeStyles-primaryInfo-5 MuiTypography-body1"
      >
        1 euro
      </p>
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root"
    >
      <p
        className="MuiTypography-root makeStyles-primaryInfo-5 MuiTypography-body1"
      >
        0.5 euro
      </p>
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root"
    >
      <button
        aria-label="expand row"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </td>
  </tr>,
  <tr
    className="MuiTableRow-root"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root"
      colSpan={12}
      style={
        Object {
          "padding": 0,
        }
      }
    />
  </tr>,
]
`;
